### ISSUE TITLE
_CheckSplitsConsistency

### ISSUE THREAD

**Issue by thibaultdvx** [aramis-lab/clinicadl#843]
Check consistency between splits (datasets and dataloaders) when validation or training on a new split.

If dataloaders differ by:
- batch_size
- sampling_weights
- shuffle
- drop_last
- collate_fn

If datasets differ by:
- type of dataset
- directory?
- datatype
- label
- transforms
- column transforms?
- data (in some cases)

Training on a new split:
- if same split: raise error;
- not necessarily the same data (no check);
- check dataset and dataloader (raise warning);
- if cannot compare, pass.

Validate:
1. takes as input 'split_idx';
2. if cannot read the ``.json``: pass dataloader via 'dataloader';
3. if dataloader passed, must be same data;
4. if dataloader passed and cannot check dataset, raise warning.
